pkg/sql/tablemetadatacache/table_metadata_updater.go

```diff
@@ -135,6 +146,21 @@ func (u *tableMetadataUpdater) upsertBatch(
 	)
 }
 
+func (u *tableMetadataUpdater) getRowsToUpdateCount(ctx context.Context) (int, error) {
+	query := `
+SELECT count(*)::INT 
+FROM system.namespace t
```

Comment:
Can we add `AS OF SYSTEM TIME follower_read_timestamp()`? We should add this clause to other queries in the job that query from `system.namespace` to avoid contention. I can file a follow-up issue to add that to the rest of the queries.

Reply:
Done